src/screens/calendar/Calendar.tsx

```diff
@@ -11,11 +11,17 @@ import { getISODateString, parseISO } from 'utils/dates'
 import { RequestsContextProvider } from 'contexts/RequestsProvider'
 import { useBooleanState } from 'hooks/useBooleanState'
 import { LoadingModal } from 'components/LoadingModal'
+import { useBackHandler } from 'hooks/useBackHandler'
+import { RouteProp, useFocusEffect, useNavigation, useRoute } from '@react-navigation/native'
+import { BottomTabRoutes } from 'navigation/types'
 import { CategoriesSlider } from './components/CategoriesSlider'
 
 const CalendarToWrap = () => {
+  const [prevScreen, setPrevScreen] = useState('')
```

Comment:
`prevScreen` shouldn't be part of component state, it's navigation responsibility.

Reply:
I need this useState to hold the prevState value as I need to clear the params.prevState and make it undefined. Without it the back handler doesn't work properly:

https://user-images.githubusercontent.com/70109479/173858555-9e2ce324-ed6d-41ec-9528-31cfea25974a.mp4

Reply:
And version with useState used to cache params.prevScreen:

https://user-images.githubusercontent.com/70109479/173858771-7ef2ea63-4ddb-4e63-b75c-4a4d4940c4d2.mp4

Reply:
Thanks for the videos. It's a bad design if components as part of the internal state keep the previous screen name :D 
Please get it directly from navigation.

Reply:
You can check if the previous screen was "NOTIFICATIONS" inside `usePrevScreenBackHandler.ts` hook